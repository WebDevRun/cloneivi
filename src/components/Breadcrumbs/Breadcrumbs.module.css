.breadcrumbs {
  display: flex;
  color: rgba(255, 255, 255, 0.56);
  font-weight: 500;
}

.item.slash {
  font-size: 0.8125rem;
  line-height: 1rem;
}
.item.slash > .link {
  color: white;
}
.item.slash > .link:active, .item.slash > .link:hover {
  background-color: #7e798f;
}
.item.slash > .link:not(:hover):not(:active) {
  box-shadow: 0 0.125rem 0 #7e798f, inset 0 0 0 #7e798f;
  transition-duration: 0.4s;
}
.item.slash > .link:hover:not(:active) {
  box-shadow: 0 0 0 #7e798f, inset 0 -1.5em 0 #7e798f;
  transition-duration: 80ms;
}
.item.dot {
  font-size: 0.9375rem;
  line-height: 1.25rem;
}
.item.slash:not(:first-child)::before {
  content: "/";
  margin-left: 0.25rem;
  margin-right: 0.1875rem;
}
.item.dot:not(:first-child)::before {
  display: inline-block;
  content: "";
  width: 0.125rem;
  height: 0.125rem;
  background-color: rgba(255, 255, 255, 0.56);
  margin-left: 0.4375rem;
  margin-right: 0.4375rem;
  margin-bottom: 0.1875rem;
}

.link {
  color: rgba(255, 255, 255, 0.72);
  transition-duration: 0.4s;
}
.link:active, .link:hover {
  color: white;
}/*# sourceMappingURL=Breadcrumbs.module.css.map */