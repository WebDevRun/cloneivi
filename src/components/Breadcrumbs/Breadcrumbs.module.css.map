{"version":3,"sources":["Breadcrumbs.module.scss","Breadcrumbs.module.css"],"names":[],"mappings":"AAEA;EACE,aAAA;EACA,gCAAA;EACA,gBAAA;ACDF;;ADKE;EACE,oBAAA;EACA,iBAAA;ACFJ;ADII;EACE,YAAA;ACFN;ADIM;EAEE,yBAAA;ACHR;ADMM;EACE,qDAAA;EACA,yBAAA;ACJR;ADOM;EACE,mDAAA;EACA,yBAAA;ACLR;ADUE;EACE,oBAAA;EACA,oBAAA;ACRJ;ADWE;EACE,YAAA;EACA,oBAAA;EACA,uBAAA;ACTJ;ADYE;EACE,qBAAA;EACA,WAAA;EACA,eAAA;EACA,gBAAA;EACA,2CAAA;EACA,sBAAA;EACA,uBAAA;EACA,wBAAA;ACVJ;;ADcA;EACE,gCAAA;EACA,yBAAA;ACXF;ADaE;EAEE,YAAA;ACZJ","file":"Breadcrumbs.module.css"}