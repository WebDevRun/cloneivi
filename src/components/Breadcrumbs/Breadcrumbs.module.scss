@use '../../styles/functions' as fn;

.breadcrumbs {
  display: flex;
  color: rgba(white, 0.56);
  font-weight: 500;
}

.item {
  &.slash {
    font-size: fn.toRem(13px);
    line-height: fn.toRem(16px);

    & > .link {
      color: white;

      &:active,
      &:hover {
        background-color: #7e798f;
      }

      &:not(:hover):not(:active) {
        box-shadow: 0 fn.toRem(2px) 0 #7e798f, inset 0 0 0 #7e798f;
        transition-duration: 0.4s;
      }

      &:hover:not(:active) {
        box-shadow: 0 0 0 #7e798f, inset 0 -1.5em 0 #7e798f;
        transition-duration: 80ms;
      }
    }
  }

  &.dot {
    font-size: fn.toRem(15px);
    line-height: fn.toRem(20px);
  }

  &.slash:not(:first-child)::before {
    content: '/';
    margin-left: fn.toRem(4px);
    margin-right: fn.toRem(3px);
  }

  &.dot:not(:first-child)::before {
    display: inline-block;
    content: '';
    width: fn.toRem(2px);
    height: fn.toRem(2px);
    background-color: rgba(white, 0.56);
    margin-left: fn.toRem(7px);
    margin-right: fn.toRem(7px);
    margin-bottom: fn.toRem(3px);
  }
}

.link {
  color: rgba(white, 0.72);
  transition-duration: 0.4s;

  &:active,
  &:hover {
    color: white;
  }
}
