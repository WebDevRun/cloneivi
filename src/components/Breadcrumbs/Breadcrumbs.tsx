import cn from 'classnames'
import Link from 'next/link'
import { FC, ReactNode } from 'react'

import styles from './Breadcrumbs.module.scss'

export type CrumbItem = {
  text: ReactNode
  path: string
}

export interface BreadcrumbsProps {
  items: CrumbItem[]
  separator: 'slash' | 'dot'
  lastAsLink?: boolean
}

export const Breadcrumbs: FC<BreadcrumbsProps> = ({
  items,
  separator,
  lastAsLink = false,
}: BreadcrumbsProps) => {
  return (
    <ul className={styles.breadcrumbs}>
      {items.map((crumb, i) => (
        <li key={i} className={cn(styles.item, styles[separator])}>
          {/** Обернуть в компонент LinkBtn */}

          {lastAsLink && (
            <Link href={crumb.path} className={styles.link}>
              {crumb.text}
            </Link>
          )}

          {!lastAsLink && i !== items.length - 1 && (
            <Link href={crumb.path} className={styles.link}>
              {crumb.text}
            </Link>
          )}
          {!lastAsLink && i === items.length - 1 && crumb.text}
        </li>
      ))}
    </ul>
  )
}
