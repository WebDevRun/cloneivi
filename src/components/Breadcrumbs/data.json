[
  {
    "text": "Home",
    "path": "/"
  },
  {
    "text": "Фёдор Бондарчук",
    "path": "/person/7982ecf2-8fae-471c-8ddf-2e3cbdab360e"
  }
]