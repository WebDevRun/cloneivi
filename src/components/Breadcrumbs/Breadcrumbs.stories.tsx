import { Meta, StoryObj } from '@storybook/react'

import '@styles/index.scss'

import { Breadcrumbs } from './Breadcrumbs'
import crumbsData from './data.json'

const meta: Meta<typeof Breadcrumbs> = {
  title: 'Breadcrumbs',
  tags: ['autodocs'],
  component: Breadcrumbs,
  argTypes: {},
}

export default meta

type Story = StoryObj<typeof Breadcrumbs>

export const SlashDefault: Story = {
  args: {
    items: crumbsData,
    separator: 'slash',
  },
}

export const Dot: Story = {
  args: {
    items: crumbsData,
    separator: 'dot',
  },
}
