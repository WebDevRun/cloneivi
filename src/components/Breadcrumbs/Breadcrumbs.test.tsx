import { render, screen } from '@testing-library/react'

import { Breadcrumbs } from './Breadcrumbs'

describe('Breadcrumbs tests', () => {
  it('', () => {})
})
