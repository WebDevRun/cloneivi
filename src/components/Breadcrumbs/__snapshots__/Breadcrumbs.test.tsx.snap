// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Breadcrumbs тесты снимок с параметрами SlashDefault 1`] = `
<ul
  className="breadcrumbs"
>
  <li
    className="item slash"
  >
    <a
      className="link"
      href="/"
      onClick={[Function]}
      onMouseEnter={[Function]}
      onTouchStart={[Function]}
    >
      Home
    </a>
  </li>
  <li
    className="item slash"
  >
    Фёдор Бондарчук
  </li>
</ul>
`;
